R/tax_spellcheck.R

```diff
@@ -0,0 +1,348 @@
+    stop("Please specify 'names' as a single column name in 'x'")
+  }
+  if(!names %in% colnames(x)) {
+    stop("'names' is not a column name in 'x'")
+  }
+  if(!is.character(x[,names])) {
+    stop("The 'names' column in 'x' must contain character data")
+  }
+  if(all(is.na(x[,names]))) {
+    stop("All taxon name entries in the specified column are NA")
+  }
+
+  # groups: If not NULL, a 1L character vector denoting a character column in x
+  if(!is.null(groups)) {
+    if(length(groups) != 1 | !is.character(groups)) {
+      stop("Please specify 'groups' as a single column name in 'x")
+    }
+    if(!groups %in% colnames(x)) {
+      stop("'groups' is not a column name in 'x'")
+    }
+    if(!is.character(x[,groups])) {
+      stop("The 'groups' column in 'x' must contain character data")
+    }
+    groups <- x[,groups]
+  } else {
+    groups <- substring(x[,names], 1, 1)
+  }
+
+  # dissim: a 1L numeric > 0 and < 1
+  if(!is.numeric(dissim) | length(dissim) != 1) {
+    stop("'dissim' must be a single numeric, greater than 0, less than 1")
+  }
+  if(dissim >= 1 | dissim <= 0) {
+    stop("'dissim' must be a single numeric, greater than 0, less than 1")
+  }
+
+  # srt: a 1L integer >= 0
+  if(!is.null(srt)) {
+    if(!is.numeric(srt) | length(srt) != 1) {
+      stop("'srt' must be a single positive integer, or zero")
+    }
+    if(srt < 0 | srt %% 1 != 0) {
+      stop("'srt' must be a single positive integer, or zero")
+    }
+  }
+
+  # end: a 1L integer >= 0
+  if(!is.null(end)) {
+    if(!is.numeric(end) | length(end) != 1) {
+      stop("'end' must be a single positive integer, or zero")
+    }
+    if(end < 0 | end %% 1 != 0) {
+      stop("'end' must be a single positive integer, or zero")
+    }
+  }
+
+  # pref: If not NULL, a character vector
+  if(!is.null(pref)) {
+    if(!is.vector(pref)) {
+      stop("'pref' must be a character vector or NULL")
+    }
+    if(!is.character(pref)) {
+      stop("'pref' must be a character vector or NULL")
+    }
+    pref <- na.omit(pref)
+  }
+
+  # suff: If not NULL, a character vector
+  if(!is.null(suff)) {
+    if(!is.vector(suff)) {
+      stop("'suff' must be a character vector or NULL")
+    }
+    if(!is.character(suff)) {
+      stop("'suff' must be a character vector or NULL")
+    }
+    suff <- na.omit(suff)
+  }
+
+  # verbose: a 1L logical vector
+  if(!is.logical(verbose) | length(verbose) != 1) {
+    stop("'verbose' must be a single logical")
+  }
+  if(length(grep("[^[[:alpha:]]", x[,names])) != 0) {
+    if(verbose) {warning("Non-letter characters present in the taxon names")}
+  }
+
+
+  # FORMAT INPUT DATA ------------------------------------------------------- #
+
+  # names data.frame, drop missing names, fill missing groups alphabetically
+  x <- x2 <- data.frame(groups = groups, names = x[,names])
+  x <- x[!duplicated(x),,drop = FALSE]
+  x <- x[!is.na(x[,"names"]),,drop = FALSE]
+  no_group <- which(is.na(x[,"groups"]))
```

Comment:
Is this working as it should? Also, we often have data (e.g. PBDB) that just have empty rows, interpreted as "" in R. We might need something else here to capture that (I experienced this issue with your brachs example).